packages/app/src/components/NetworkPropertyPanel.tsx

```diff
@@ -0,0 +1,343 @@
+import React, { useMemo } from 'react';
+import Link from 'next/link';
+import { pickBy } from 'lodash';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import { JSONTree } from 'react-json-tree';
+import {
+  Accordion,
+  Box,
+  Button,
+  CopyButton,
+  TableData,
+  Text,
+} from '@mantine/core';
+import { notifications } from '@mantine/notifications';
+
+import HyperJson from '@/components/HyperJson';
+import { Table } from '@/components/Table';
+import {
+  headerColumns,
+  networkColumns,
+  SectionWrapper,
+} from '@/LogSidePanelElements';
+import { CollapsibleSection } from '@/LogSidePanelElements';
+import { CurlGenerator } from '@/utils/curlGenerator';
+
+interface NetworkPropertyPanelProps {
+  eventAttributes: Record<string, any>;
+  onPropertyAddClick?: (key: string, value: string) => void;
+  generateSearchUrl: (query?: string, timeRange?: [Date, Date]) => string;
+}
+
+// https://github.com/reduxjs/redux-devtools/blob/f11383d294c1139081f119ef08aa1169bd2ad5ff/packages/react-json-tree/src/createStylingFromTheme.ts
+const JSON_TREE_THEME = {
+  base00: '#00000000',
+  base01: '#383830',
+  base02: '#49483e',
+  base03: '#75715e',
+  base04: '#a59f85',
+  base05: '#f8f8f2',
+  base06: '#f5f4f1',
+  base07: '#f9f8f5',
+  base08: '#f92672',
+  base09: '#fd971f',
+  base0A: '#f4bf75',
+  base0B: '#a6e22e',
+  base0C: '#a1efe4',
+  base0D: '#8378FF', // Value Labels
+  base0E: '#ae81ff',
+  base0F: '#cc6633',
+};
+
+const parseHeaders = (
+  keyPrefix: string,
+  parsedProperties: any,
+): [string, string][] => {
+  const reqHeaderObj: Record<
+    string,
+    string | string[] | Record<string, string>
+  > = pickBy(parsedProperties, (value, key) => key.startsWith(keyPrefix));
+
+  return Object.entries(reqHeaderObj).flatMap(([fullKey, _value]) => {
+    let value = _value;
+    try {
+      if (typeof _value === 'string') {
+        value = JSON.parse(_value);
+      }
+    } catch (e) {
+      // ignore
+    }
+
+    // Replacing _ -> - is part of the otel spec, idk why
+    const key = fullKey.replace(keyPrefix, '').replace('_', '-');
+
+    let keyVal = [[key, `${value}`]] as [string, string][];
+
+    if (Array.isArray(value)) {
+      keyVal = value.map(value => [key, `${value}`] as [string, string]);
+    } else if (typeof value === 'object' && Object.keys(value).length > 0) {
+      try {
+        // TODO: We actually shouldn't be re-serializing this as it may mess with the original value
+        keyVal = [[key, `${JSON.stringify(value)}`]] as [string, string][];
```

Comment:
if object `value` can only from parsing, we can use `_value` at here

Reply:
I think you're right, I'm not sure why we did this to begin with, this code is migrated straight from v1 though and hasn't had any issues in production (IIRC) so just leaving this one as is.